src/httpd_dacp.c

```diff
@@ -783,12 +785,42 @@ dacp_reply_cue_play(struct evhttp_request *req, struct evbuffer *evbuf, char **u
 	DPRINTF(E_LOG, L_DACP, "Invalid index (%s) in cue request\n", param);
     }
 
-  /* If selection was from Up Next queue (command will be playnow), then index is relative */
-  //TODO playnow for mode = 1 (index is relativ to the history queue
+  /* If selection was from Up Next queue or history queue (command will be playnow), then index is relative */
+  hist = 0;
   if ((param = evhttp_find_header(query, "command")) && (strcmp(param, "playnow") == 0))
-    pos += status.pos_pl;
+    {
+      /* If mode parameter is -1, the index is relative to the history queue, otherwise to the Up Next queue */
+      param = evhttp_find_header(query, "mode");
+      if (param && strcmp(param, "-1") == 0)
+	{
+	  /* Play from history queue */
+	  hist = 1;
+	  history = player_history_get();
+	  if (history->count > pos)
+	    {
+	      pos = history->id[(history->start_index + history->count - pos - 1) % MAX_HISTORY_COUNT];
```

Comment:
Could you split this in two lines, so first calculate the id and then find the pos?